modyn/trainer_server/internal/trainer/remote_downsamplers/remote_craig_downsampling.py

```diff
@@ -43,13 +46,16 @@ def __init__(
 
         self.selection_batch = params_from_selector["selection_batch"]
         self.greedy = params_from_selector["greedy"]
+        self.full_grad_approximation = params_from_selector["full_grad_approximation"]
+        assert self.full_grad_approximation in FULL_GRAD_APPROXIMATION
+
         if self.greedy not in OPTIMIZER_CHOICES:
             raise ValueError(
                 f"The required Greedy optimizer is not available. Pick one of the following: {OPTIMIZER_CHOICES}"
             )
 
-        # This class uses the embedding recorder
-        self.requires_coreset_supporting_module = True
+        self.requires_coreset_supporting_module = self.full_grad_approximation == "LastLayerWithEmbedding"
+        self.requires_data_label_by_label = True
```

Comment:
Craig always selects points label by label. Previously this field is set implicitly to True in the base class `AbstractPerLabelRemoteDownsamplingStrategy`